src/celeritas/mat/MaterialParams.cc

```diff
@@ -137,8 +174,10 @@ MaterialParams::MaterialParams(Input const& inp)
 
     CELER_ENSURE(this->data_);
     CELER_ENSURE(this->host_ref().elements.size() == inp.elements.size());
+    CELER_ENSURE(this->host_ref().isotopes.size() == inp.isotopes.size());
     CELER_ENSURE(this->host_ref().materials.size() == inp.materials.size());
     CELER_ENSURE(el_labels_.size() == inp.elements.size());
+    CELER_ENSURE(isot_labels_.size() == inp.isotopes.size());
```

Comment:
order the `ensure`s so isotopes are before elements (so that all the construction/data is always ordered from finest to coarsest)